tslearn/nonmyopic.py

```diff
@@ -0,0 +1,370 @@
+from sklearn.metrics import confusion_matrix
+from sklearn.base import BaseEstimator, ClassifierMixin
+from tslearn.utils import to_time_series_dataset
+from tslearn.clustering import TimeSeriesKMeans
+import numpy as np
+from scipy.sparse import coo_matrix
+
+
+class NonMyopicEarlyClassification(BaseEstimator, ClassifierMixin):
+    """Early Classification modelling for time series using the model presented in [1]
+
+    Parameters
+    ----------
+    n_clusters : int
+        Number of clusters to form.
+
+    solver :
+        The method of gradient descent used during training
+
+    hidden_layer_sizes : tuple of int
+        Size of the hidden layers
+
+    random_state : int
+        Seed of the MLP
+
+    maximum_iteration : int
+        Maximum number of iterations performed by the MLP
+
+    minimum_time_stamp : int
+        Earliest time at which a classification can be performed on a time series
+
+    lamb : float
+        Value of the hyper parameter lambda used during the computation of the cost function to evaluate the probability
```

Comment:
Your lines should never be longer than 79 characters, otherwise PEP8 checks will fail.